Civi/Api4/Service/Spec/SpecGatherer.php

```diff
@@ -97,7 +97,7 @@ private function addDAOFields(string $entityName, string $action, RequestSpec $s
       if (!empty($DAOField['component']) && !\CRM_Core_Component::isEnabled($DAOField['component'])) {
         continue;
       }
-      if ($DAOField['name'] == 'is_active' && empty($DAOField['default'])) {
+      if ($DAOField['name'] == 'is_active' && (!isset($DAOField['default']) || $DAOField['default'] === '')) {
         $DAOField['default'] = '1';
       }
```

Comment:
Oof, I forgot this code existed. I think we should just delete it. Our schema used to be very inconsistent but there's been a lot of work lately to clean up boolean columns and ensure they always have the correct default set at the sql level so I think we can rely on that now and get rid of this.

Reply:
OK. What about the test?
It seems a bit redundant after deletion, but more test is usually better than less :smiley:

Reply:
Yes keep it. Always good to have a test!

Reply:
Removed. I've squashed the commits for nicer history.

Reply:
Perfect. Let's see what the test suite thinks of this.

Reply:
Looks like test suite doesn't like it...
I believe the reason is a missing `<default>1</default>` on civicrm_acl.is_active.
I've added a quick hack to simulate that and to verify tests pass with that.

Reply:
Quick hack didn't work so I did the proper way :smile: 
Missing default on `civicrm_acl.is_active` is just _part_ of the reasons... Though there are 2 less failing test cases.

Seems, we need to go through the rest of the failing tests.
I think it's better to divide this PR into smaller parts and extract missing default on `civicrm_acl.is_active` as it's independent of this and needs to fixed anyway.